File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = []
        for i, c in enumerate(s):
            if c in "aeiouAEIOU":
                vowels.append(c)
        vowels = deque(sorted(vowels))
        for i, c in enumerate(s):
            if c in 'aeiouAEIOU':
                s = s[0:i:] + vowels.popleft() + s[i+1::]
        return s

File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var sortVowels = function(s) {
    const vowels = []
    for (const c of s) {
        if ('aeiouAEIOU'.includes(c)) {
            vowels.push(c)
        }
    }
    vowels.sort((a, b) => a < b ? 1 : -1)
    for (let i=0; i < s.length; i++) {
        if ('aeiouAEIOU'.includes(s[i])) {
            s = s.substring(0, i) + vowels.pop() + s.substring(i+1)
        }
    }
    return s
};